pxr/usdImaging/lib/usdImagingGL/engine.cpp

```diff
@@ -637,6 +641,30 @@ UsdImagingGLEngine::GetPrimPathFromInstanceIndex(
                                              instanceContext);
 }
 
+SdfPath
+UsdImagingGLEngine::ConvertCachePathToIndexPath(SdfPath const& cachePath)
+{
+    if (ARCH_UNLIKELY(_legacyImpl)) {
+        return SdfPath();
+    }
+
+    TF_VERIFY(_delegate);
+
+    return _delegate->ConvertCachePathToIndexPath(cachePath);
+}
```

Comment:
Instead of two different checks, can we just do a 
```
if (ARCH_LIKELY(_delegate)){
   return _delegate->ConvertCachePathToIndexPath(cachePath);
}

return SdfPath();
```

and similarly below?

Reply:
Sure - if it's kosher to assume that having _delegate be non-zero implies _legacyImpl is zero, then combining into one check makes sense.  We do lose the "warning" / feedback we get from TF_VERIFY when both are zero, however.

It's worth noting, though, that I was following the template of GetPrimPathFromInstanceIndex:

https://github.com/PixarAnimationStudios/USD/blob/dev/pxr/usdImaging/lib/usdImagingGL/engine.cpp#L629

...and this pattern is used by many other methods, as well  - ie, SetRootTransform, SetRootVisibility, etc.

Do you want me to go through and clean up all these as well? Perhaps in a different PR? If so, I assume we can make similar assumptions about the relationship between _taskController and _legacyImpl?